cs/src/Connections/ConnectionStatusChangedEventArgs.cs

```diff
@@ -0,0 +1,40 @@
+﻿// <copyright file="ConnectionStatusChangedEventArgs.cs" company="Microsoft">
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.VsSaaS.TunnelService;
+
+/// <summary>
+/// Connection status change event args.
+/// </summary>
+public class ConnectionStatusChangedEventArgs : EventArgs
+{
+    /// <summary>
+    /// Create a new instance of <see cref="ConnectionStatusChangedEventArgs"/>.
+    /// </summary>
+    public ConnectionStatusChangedEventArgs(ConnectionStatus previousStatus, ConnectionStatus status, Exception? disconnectException)
+    {
+        PreviousStatus = previousStatus;
```

Comment:
Is there a reason we dont set disconnectException here?

Reply:
No reason. Will fix. Good catch.